llvm/test/CodeGen/NVPTX/i128-ld-st.ll

```diff
@@ -0,0 +1,28 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc < %s -O0 -mcpu=sm_20 | FileCheck %s
+; RUN: %if ptxas %{ llc < %s -O0 -mcpu=sm_20 | %ptxas-verify %}
+
+target triple = "nvptx64-nvidia-cuda"
+
+define i128 @foo() {
+; CHECK-LABEL: foo(
+; CHECK:       {
+; CHECK-NEXT:    .reg .b64 %rd<3>;
+; CHECK-EMPTY:
+; CHECK-NEXT:  // %bb.0: // %entry
+; CHECK-NEXT:    bra.uni $L__BB0_1;
+; CHECK-NEXT:  $L__BB0_1: // %while.cond
+; CHECK-NEXT:    // =>This Inner Loop Header: Depth=1
+; CHECK-NEXT:    mov.b64 %rd1, 0;
+; CHECK-NEXT:    ld.u8 %rd2, [%rd1];
+; CHECK-NEXT:    st.v2.u64 [%rd1], {%rd2, %rd1};
+; CHECK-NEXT:    bra.uni $L__BB0_1;
+entry:
```

Comment:
the test case could be cleaned up a bit, dropping the loop and using valid pointers, so there are no questions of whether the input is valid.

E.g. https://godbolt.org/z/EsvbY3dav